Source/Core/Common/HttpRequest.cpp

```diff
@@ -173,6 +180,17 @@ void HttpRequest::Impl::FollowRedirects(long max)
   curl_easy_setopt(m_curl.get(), CURLOPT_MAXREDIRS, max);
 }
 
+std::string HttpRequest::Impl::GetHeaderValue(std::string_view name) const
+{
+  for (const auto& [key, value] : m_response_headers)
```

Comment:
I think a slight optimization could be done here?  If `m_response_headers` supported heterogeneous lookup you could use the `name` as the key in the map.  I don't think this is blocking but something to consider.